From: Aquila Macedo <aquilamacedo@riseup.net>
Date: Wed, 20 Sep 2023 23:36:35 -0300
Subject: Fix spelling error in openvpn binary
Forwarded: yes, https://github.com/OpenVPN/openvpn/pull/433
---
 src/openvpn/options.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/openvpn/options.c b/src/openvpn/options.c
index 5194e7d..c795f57 100644
--- a/src/openvpn/options.c
+++ b/src/openvpn/options.c
@@ -601,7 +601,7 @@ static const char usage_message[] =
     "                  Windows Certificate System Store.\n"
 #endif
     "--tls-cipher l  : A list l of allowable TLS ciphers separated by : (optional).\n"
-    "--tls-ciphersuites l: A list of allowed TLS 1.3 cipher suites seperated by : (optional)\n"
+    "--tls-ciphersuites l: A list of allowed TLS 1.3 cipher suites separated by : (optional)\n"
     "                : Use --show-tls to see a list of supported TLS ciphers (suites).\n"
     "--tls-cert-profile p : Set the allowed certificate crypto algorithm profile\n"
     "                  (default=legacy).\n"
