Description: Segfault in check_session_buf_not_used
Origin: https://github.com/OpenVPN/openvpn/issues/449

diff --git a/src/openvpn/ssl.c b/src/openvpn/ssl.c
index 210a70165..9c88b9e84 100644
--- a/src/openvpn/ssl.c
+++ b/src/openvpn/ssl.c
@@ -3258,6 +3258,11 @@ check_session_buf_not_used(struct buffer *to_link, struct tls_session *session)
     for (int i = 0; i < KS_SIZE; i++)
     {
         struct key_state *ks = &session->key[i];
+        if (ks->state == S_UNDEF)
+        {
+            continue;
+        }
+
         for (int j = 0; j < ks->send_reliable->size; j++)
         {
             if (ks->send_reliable->array[i].buf.data == dataptr)
